src/backend/distributed/connection/remote_commands.c

```diff
@@ -75,6 +75,90 @@ ForgetResults(MultiConnection *connection)
 
 
 /*
+ * NonblockingForgetResults clears a connection from pending activity if doing
+ * so does not require network IO. Returns true if successful, false
+ * otherwise.
+ */
+bool
+NonblockingForgetResults(MultiConnection *connection)
+{
+	PGconn *pgConn = connection->pgConn;
+	bool wasNonblocking = PQisnonblocking(pgConn);
+
+	if (PQstatus(pgConn) != CONNECTION_OK)
+	{
+		return false;
+	}
+
+	while (true)
+	{
+		PGresult *result = NULL;
+
+		CHECK_FOR_INTERRUPTS();
```

Comment:
Yes, I think we need to. Consider the case that the other side is sending results faster than we can process them...